src/joyApi.ts

```diff
@@ -371,6 +371,42 @@ export class JoyApi {
     };
   }
 
+
+  async calculateCirculatingSupply() {
+    const VESTING_STRING_HEX = "0x76657374696e6720";
+
+    const accounts = [];
+    const amounts: BN[] = [];
+    const lockData = await this.api.query.balances.locks.entries();
+
+    for (let [storageKey, palletBalances] of lockData) {
+      let vested = new BN(0);
+      for (let palletBalance of palletBalances) {
+        if (
+          palletBalance.id.toString() === VESTING_STRING_HEX &&
+          palletBalance.amount.toBn().gt(vested)
+        ) {
+          vested = palletBalance.amount.toBn();
+        }
+      }
+
+      if (vested.gtn(0)) {
```

Comment:
Probably safer to not use any of the `n` functions..
```suggestion
      if (!vested.isZero()) {
```

Reply:
or simply `if (vested.gt(new BN(0))) {`